### DISCUSSION TITLE
Failed recognition of __hash__ function from parent model which inherited from BaseModel.

### DISCUSSION THREAD

**Discussion by ghost** [pydantic/pydantic#8868] [Question]
Hi, I'm wondering if I'm using the BaseModel wrong.

I have a setup where the parent model has a hash function and I want the child model to inherit it, similar to this one
```py
from pydantic import BaseModel

class MyBaseModel(BaseModel):
    def __hash__(self):
        assert hasattr(self, "foo")
        return hash(self.foo)

class Foo(MyBaseModel):
    foo: str = "foo"

s = {Foo()}
print(s, hash(Foo()) == hash("foo"))
# {Foo(foo='foo')} True
```

And this runs perfectly fine, however pyright does not seem to recognize that the child model inherited the hash function, as it throws an "error: Set entry must be hashable". I have verified that removing the BaseModel inheritance fixes the pyright error and thus I assume that the issue lies within pydantic.

This is running on
* pydantic version: 2.6.1
* python version: 3.12.3 
* pyright version: 1.1.349

Any help would be greatly appricated.

**Comment by ishefi** [comment]
According to [this](https://github.com/microsoft/pylance-release/issues/3936), this is indeed a bug in pydantic

**Comment by JensWehner** [comment]
I am curious as I have run into the same error, if there is any chance to fix this?
